worlds/mm2/__init__.py

```diff
@@ -0,0 +1,314 @@
+                if region in etank_1ups:
+                    stage.add_locations(etank_1ups[region], MM2Location)
+            if self.options.consumables in (Consumables.option_weapon_health,
+                                            Consumables.option_all):
+                if region in energy_pickups:
+                    stage.add_locations(energy_pickups[region], MM2Location)
+            self.multiworld.regions.append(stage)
+
+    def create_item(self, name: str, force_non_progression: bool = False) -> MM2Item:
+        item = item_table[name]
+        classification = ItemClassification.filler
+        if item.progression and not force_non_progression:
+            classification = ItemClassification.progression_skip_balancing \
+                if item.skip_balancing else ItemClassification.progression
+        if item.useful:
+            classification |= ItemClassification.useful
+        return MM2Item(name, classification, item.code, self.player)
+
+    def get_filler_item_name(self) -> str:
+        return self.multiworld.random.choices(list(filler_item_weights.keys()),
+                                              weights=list(filler_item_weights.values()))[0]
+
+    def create_items(self) -> None:
+        itempool = []
+        # grab first robot master
+        robot_master = self.item_id_to_name[0x880101 + self.options.starting_robot_master.value]
+        self.multiworld.push_precollected(self.create_item(robot_master))
+        itempool.extend([self.create_item(name) for name in stage_access_table.keys()
+                         if name != robot_master])
+        itempool.extend([self.create_item(name) for name in robot_master_weapon_table.keys()])
+        itempool.extend([self.create_item(name) for name in item_item_table.keys()])
+        total_checks = 24
+        if self.options.consumables in (Consumables.option_1up_etank,
+                                        Consumables.option_all):
+            total_checks += 20
+        if self.options.consumables in (Consumables.option_weapon_health,
+                                        Consumables.option_all):
+            total_checks += 27
+        remaining = total_checks - len(itempool)
+        itempool.extend([self.create_item(name)
+                         for name in self.multiworld.random.choices(list(filler_item_weights.keys()),
+                                                                    weights=list(filler_item_weights.values()),
+                                                                    k=remaining)])
+        self.multiworld.itempool += itempool
+
+    set_rules = set_rules
+
+    def generate_early(self) -> None:
+        if (not self.options.yoku_jumps
+            and self.options.starting_robot_master.current_key == "heat_man") or \
+                (not self.options.enable_lasers
+                 and self.options.starting_robot_master.current_key == "quick_man"):
```

Comment:
You don't need `.current_key` with a `Choice` option comparing to a string
```suggestion
            and self.options.starting_robot_master == "heat_man") or \
                (not self.options.enable_lasers
                 and self.options.starting_robot_master == "quick_man"):
```